controllers/stats/EditorialReportComponentHandler.inc.php

```diff
@@ -0,0 +1,239 @@
+
+	/** @var array|null Configuration assoc array for available filters */
+	public $_filters = null;
+
+	/** @var array Localized strings to pass to the component */
+	public $_i18n = [];
+
+	/** @var array List of submission stages and their statistics */
+	public $_submissionsStage = [];
+
+	/** @var array Data to present on the pie chart */
+	public $_editorialChartData = [];
+
+	/**
+	 * Constructor
+	 *
+	 * @param $apiUrl string The URL to fetch stats from
+	 * @param $args array Optional arguments
+	 */
+	public function __construct($apiUrl, $args = array()) {
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_MANAGER);
+		AppLocale::requireComponents(LOCALE_COMPONENT_APP_MANAGER);
+		parent::__construct();
+
+		$this->_apiUrl = $apiUrl;
+		$this->init($args);
+	}
+
+	/**
+	 * Given an array with statistics extracts the required editorial information to display in the component
+	 * @param $rangedStatistics array An array with a subset of the results from a call to PKPStatsService::getSubmissionStatistics
+	 * @param $statistics array An array with the results from a call to PKPStatsService::getSubmissionStatistics
+	 */
+	public static function extractEditorialStatistics($rangedStatistics, $statistics) {
+		$editorialStatistics = [];
+		$percentageField = 1;
+		$averageField = 2;
+		foreach ([
+			'SUBMISSION_RECEIVED' => [__('manager.statistics.editorial.submissionsReceived')],
+			'SUBMISSION_ACCEPTED' => [__('manager.statistics.editorial.submissionsAccepted')],
+			'SUBMISSION_DECLINED_TOTAL' => [__('manager.statistics.editorial.submissionsDeclined')],
+			'SUBMISSION_DECLINED_INITIAL' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.deskReject')],
+			'SUBMISSION_DECLINED' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.postReview')],
+			'SUBMISSION_DECLINED_OTHER' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.other')],
+			'SUBMISSION_PUBLISHED' => [__('manager.statistics.editorial.submissionsPublished')],
+			'SUBMISSION_DAYS_TO_FIRST_DECIDE' => [__('manager.statistics.editorial.averageDaysToDecide'), $averageField],
+			'SUBMISSION_DAYS_TO_ACCEPT' => ['&emsp;' . __('manager.statistics.editorial.averageDaysToAccept'), $averageField],
+			'SUBMISSION_DAYS_TO_REJECT' => ['&emsp;' . __('manager.statistics.editorial.averageDaysToReject'), $averageField],
+			'SUBMISSION_ACCEPTANCE_RATE' => [__('manager.statistics.editorial.acceptanceRate'), $percentageField],
+			'SUBMISSION_REJECTION_RATE' => [__('manager.statistics.editorial.rejectionRate'), $percentageField],
+			'SUBMISSION_DECLINED_INITIAL_RATE' => ['&emsp;' . __('manager.statistics.editorial.deskRejectRate'), $percentageField],
+			'SUBMISSION_DECLINED_RATE' => ['&emsp;' . __('manager.statistics.editorial.postReviewRejectRate'), $percentageField],
+			'SUBMISSION_DECLINED_OTHER_RATE' => ['&emsp;' . __('manager.statistics.editorial.otherRejectRate'), $percentageField]
+		] as $field => list($name, $type)) {
```

Comment:
I'd rather keep it this way, to keep it short and avoid duplicating hard coded strings, but if you think it's better to simplify, I can come up with something more clean.

For example, the average fields should have a "source", which is `'AVG_' + FIELD_NAME`, such hard coded things are better to be kept hard coded instead of "composed" to avoid failed magical replaces :D

Reply:
> I'd rather leave it this way, to keep it short and avoid duplicating hard coded strings
:+1: I've only now noticed that they are strings and not `CONSTANTS`. I think Alec will probably want constants to be defined, to avoid magic strings. For consistency, it is probably best to make these into constants and then use the constants rather than the strings, even if it seems like unnecessary duplication.

>  Instead of concatenating, it's better to leave such hard coded things "integral", to avoid missing entries in magical replaces/refactorings

:100:

Reply:
I've got the message. I'll drop some time to make things more abstract.